buildSrc/src/main/groovy/airbyte-docker.gradle

```diff
@@ -0,0 +1,216 @@
+                    logger.debug "did match"
+                    return false
+                }
+            }
+
+            if(notUpToDate) {
+                return false;
+            }
+
+            logger.debug "stored " + stored.trim()
+
+            def upToDate = current.equals(stored.trim())
+
+            logger.debug "uptodate " + upToDate.toString()
+
+            return upToDate
+        } else {
+            return false
+        }
+    }
+
+    static def createTask(Project project, String taskName, String dockerFile) {
+        if (project.file(dockerFile).exists()) {
+            def filteredProjectFiles = project.fileTree(project.projectDir).filter {
+                file -> !file.toString().contains(".venv")
+            }
+            project.task(taskName, type: AirbyteDockerTask) {
+                def dockerPath = Paths.get(project.projectDir.absolutePath, dockerFile)
+                def hash = MessageDigest.getInstance("MD5").digest(dockerPath.getBytes()).encodeHex().toString()
+                rootDir = project.rootProject.rootDir
+                projectDir = project.projectDir
+                projectFiles = filteredProjectFiles
+                idFileOutput = project.file(Paths.get(project.rootProject.rootDir.absolutePath, '.dockerversions', hash).toString())
+                baseImageHashes = getBaseImageHashes(project.rootProject.imageToVersion, dockerPath.toFile())
+                dependsOn project.assemble
+
+                outputs.upToDateWhen {
+                    return isUpToDate(logger, idFileOutput, project, dockerPath)
+                }
+            }
+        } else {
+            project.task(taskName) {
+                logger.info "Skipping ${taskName} because ${dockerFile} does not exist."
+            }
+        }
+    }
+
+    void apply(Project project) {
+        // set (and cache) global image to version map
+        project.rootProject.ext.imageToVersion = {
+            if(!project.rootProject.hasProperty("imageToVersion")) {
+                def imageToVersion = [:]
+                def stdout = new ByteArrayOutputStream()
+                project.exec {
+                    commandLine "docker", "images", "--no-trunc", "-f", "dangling=false", "--format", "{{.Repository}}:{{.Tag}} {{.ID}}"
+                    standardOutput = stdout;
+                }
+
+                "$stdout".eachLine { line ->
+                    def splits = line.split()
+                    imageToVersion.put(splits[0], splits[1])
+                }
+
+                return imageToVersion
+            } else {
+                return project.rootProject.imageToVersion
+            }
+        }()
+
+        createTask(project, 'airbyteDockerBuildMain', 'Dockerfile')
+        createTask(project, 'airbyteDockerBuildTest', 'Dockerfile.test')
+
+        project.task('airbyteDocker') {
+            dependsOn project.airbyteDockerBuildMain
+            dependsOn project.airbyteDockerBuildTest
```

Comment:
this is an integration test image, I am not convinced it should be built as part of the `build`, I think it should be build as part of the integrationTest task